tests/test_datetime.py

```diff
@@ -0,0 +1,340 @@
+# Copyright (c) QuantCo 2023-2025
+# SPDX-License-Identifier: BSD-3-Clause
+
+import operator
+from typing import get_args
+
+import numpy as np
+import pytest
+
+import ndonnx as ndx
+from ndonnx import from_numpy_dtype
+from ndonnx._typed_array.datetime import Unit
+
+from .utils import assert_equal_dtype_shape
+
+_NAT_SENTINEL = np.iinfo(np.int64).min
+
+
+@pytest.mark.parametrize("unit", get_args(Unit))
+@pytest.mark.parametrize("cls", [ndx.DateTime64DType, ndx.TimeDelta64DType])
+def test_dtype_constructor_valid_units(unit, cls):
+    cls(unit=unit)
+
+
+@pytest.mark.parametrize("cls", [ndx.DateTime64DType, ndx.TimeDelta64DType])
+def test_dtype_constructor_invalid_unit(cls):
+    with pytest.raises(TypeError):
+        cls(unit="foo")
+
+
+def test_value_prop_datetime():
+    arr = ndx.asarray(np.asarray([1, 2])).astype(ndx.DateTime64DType("s"))
+    np.testing.assert_equal(
+        arr.unwrap_numpy(), np.asarray([1, 2], dtype="datetime64[s]")
+    )
+
+
+def test_arithmetic():
+    arr_np = np.array([1, 2, "NaT"], "datetime64[s]")
+    arr = ndx.asarray(arr_np)
+    one_s_td = (arr + 1) - arr
+    one_s_td_np = (arr_np + 1) - arr_np
+
+    assert one_s_td.dtype == ndx.TimeDelta64DType("s")
+    np.testing.assert_array_equal(one_s_td.unwrap_numpy(), one_s_td_np, strict=True)
+
+
+@pytest.mark.parametrize("unit", ["s", "ns"])
+@pytest.mark.parametrize("ty", ["datetime64", "timedelta64"])
+def test_datetime_from_np_array(ty, unit):
+    np_arr = np.array([0, 1], f"{ty}[{unit}]")
+    arr = ndx.asarray(np_arr)
+    np.testing.assert_array_equal(arr.unwrap_numpy(), np_arr, strict=True)
+
+
+@pytest.mark.parametrize(
+    "scalar, dtype, res_dtype",
+    [
+        # TODO: Implement other units
```

Comment:
We have more units now?

Reply:
I parametrized many of the tests over all units now.